src/allmydata/util/connection_status.py

```diff
@@ -0,0 +1,92 @@
+import time
+from zope.interface import implementer
+from ..interfaces import IConnectionStatus
+
+@implementer(IConnectionStatus)
+class ConnectionStatus:
+    def __init__(self, connected, summary,
+                 last_connection_description, last_connection_time,
+                 last_received):
+        self._connected = connected
+        self._summary = summary
+        self._last_connection = last_connection_description
+        self._last_connection_time = last_connection_time
+        self._last_received = last_received
+
+    def is_connected(self):
+        return self._connected
+    def summarize_last_connection(self):
+        return self._summary
+    def when_established(self):
+        return self._last_connection_time
```

Comment:
(see note above). e.g. could call this API "last_connection_time" instead of "when_established" (the latter kind-of implying it returns a `Deferred` like other uses in Tahoe + Twisted)